### PR TITLE
u3d/install: display more information during sanization, including what would happen (fixes #50)

### LINKED ISSUE #50: Document installation path sanitization
Feedback:

> When I run u3d list it first told me that it needed to move my 20 unity installations. Where does it
> want to move them and why? They are currently all residing in the Applications folder as
> "Unity5_3_8f2" and so on up to and including "Unity2017_1_f3". I couldn't find anywhere in the
> readme where it wanted them to go.

Let's improve this:

- [ ] update the README with why and how path sanitization works, interactive mode etc.
- [ ] update inline help
- [ ] maybe update u3d to point to README

### PR DESCRIPTION
**Pull request by lacostej**

### CODE REVIEW

File: lib/u3d/installer.rb [hunk 1]
```diff
@@ -327,20 +337,26 @@ def install_sh(file, installation_path: nil)
...
 
   class WindowsInstaller
-    def sanitize_install(unity)
+    def sanitize_install(unity, dry_run: false)
       source_path = File.expand_path(unity.path)
       parent = File.expand_path('..', source_path)
       new_path = File.join(parent, UNITY_DIR % unity.version)
-      UI.important "Moving #{source_path} to #{new_path}..."
+
+      UI.important "Moving #{source_path} to #{new_path}..." unless dry_run
```
Comment (lacostej) [inline lib/u3d/installer.rb]: @niezbop this log is displayed before the conversion to windows paths (as opposed to what we do on oher platforms). Maybe we can move it back into the `if`. I can't really test on windows. Maybe you can?

Comment (niezbop) [inline lib/u3d/installer.rb]: Willd do

Comment (niezbop) [inline lib/u3d/installer.rb]: Just tested your branch, and the path is already Windows-ed (backslashed)!

Output from `u3d list`:
```shell
C:\WINDOWS\system32>u3d list
u3d can optionally standardize the existing Unity3d installation names and locations. See the documentation for more information.
"C:\Program Files\Unity 5.6.0f3" would move to "C:\Program Files\Unity_5.6.0f3"
1 Unity installation will be moved. Proceed?? (y/n)
y
Moving C:/Program Files/Unity 5.6.0f3 to C:/Program Files/Unity_5.6.0f3...
$ move "C:\Program Files\Unity 5.6.0f3" "C:\Program Files\Unity_5.6.0f3"
No pty implementation found. Falling back to popen. Output might be buffered
Successfully moved "C:\Program Files\Unity 5.6.0f3" to "C:\Program Files\Unity_5.6.0f3"
Version 5.6.0f3                       (C:/Program Files/Unity_5.6.0f3)
Version 5.6.1f1                       (C:/Program Files/Unity_5.6.1f1)
Version 2017.1.0f1                    (C:/Program Files/Unity_2017.1.0f1)
```

Comment (niezbop) [inline lib/u3d/installer.rb]: The log is indeed Unix style, but all the surrounding are not:

```shell
C:\WINDOWS\system32>u3d list
u3d can optionally standardize the existing Unity3d installation names and locations. See the documentation for more information.
"C:\Program Files\Unity 5.6.0f3" would move to "C:\Program Files\Unity_5.6.0f3"
1 Unity installation will be moved. Proceed?? (y/n)
y
Moving C:/Program Files/Unity 5.6.0f3 to C:/Program Files/Unity_5.6.0f3...
$ move "C:\Program Files\Unity 5.6.0f3" "C:\Program Files\Unity_5.6.0f3"
No pty implementation found. Falling back to popen. Output might be buffered
Successfully moved "C:\Program Files\Unity 5.6.0f3" to "C:\Program Files\Unity_5.6.0f3"
Version 5.6.0f3                       (C:/Program Files/Unity_5.6.0f3)
Version 5.6.1f1                       (C:/Program Files/Unity_5.6.1f1)
Version 2017.1.0f1                    (C:/Program Files/Unity_2017.1.0f1)
```

Comment (niezbop) [inline lib/u3d/installer.rb]: So maybe we should try to make it Windows style?

**Comment by lacostej** [comment]
Windows version is incomplete, mostly because I can't test it here. I think I'll merge this as is, with a follow up PR from your side to look at the Windows tests when you have time. Hopefully the implementation isn't broken :)

Curent behaviour on Mac:
image

**Pull request merged**